### PR TITLE
CSM: Document forgotten functions

### PR DESCRIPTION
**Pull request by SmallJoker**
For issue #5570

The only not documented function in CSM and the server modding is `set_last_run_mod`, which should not be used by mods.

### CODE REVIEW

File: doc/client_lua_api.md [hunk 1]
```diff
@@ -621,6 +628,8 @@ Minetest namespace reference
   version entirely. To check for the presence of engine features, test
   whether the functions exported by the wanted features exist. For example:
   `if minetest.nodeupdate then ... end`.
+* `minetest.get_builtin_path()`
+    * Returns the path to builtin. Not garantueed to return the absolute path.
```
Comment (bigfoot547) [inline doc/client_lua_api.md]: garantueed => guaranteed

File: doc/client_lua_api.md [hunk 2]
```diff
@@ -590,17 +590,24 @@ Helper functions
...
+* `string.split(str, separator=",", include_empty=false, max_splits=-1, sep_is_pattern=false)`
     * If `max_splits` is negative, do not limit splits.
     * `sep_is_pattern` specifies if separator is a plain string or a pattern (regex).
     * e.g. `string:split("a,b", ",") == {"a","b"}`
 * `string:trim()`
     * e.g. `string.trim("\n \t\tfoo bar\t ") == "foo bar"`
+* `minetest.splittext(str, limit)`: returns a string
+    * Adds new lines to the string to keep it within the specified character limit.
+    * limit: Maximal amount of characters in one line
+* `minetest.pos_to_string({x=X,y=Y,z=Z}, decimal_places))`: returns `"(X,Y,Z)"`
```
Comment (bigfoot547) [inline doc/client_lua_api.md]: I see the quotes but maybe say returns string.

File: doc/client_lua_api.md [hunk 3]
```diff
@@ -683,6 +694,12 @@ Call these functions only at load time!
 * `minetest.after(time, func, ...)`
     * Call the function `func` after `time` seconds, may be fractional
     * Optional: Variable number of arguments that are passed to `func`
+* `minetest.get_us_time()`
+    * Returns time with microsecond precision. May not return wall time.
+* `minetest.get_day_count()`
+    * Returns number days elapsed since world was created, accounting for time changes.
+* `minetest.get_timeofday()`
+    * Returns the time of day: `0` for midnight, `0.5` for midday
```
Comment (bigfoot547) [inline doc/client_lua_api.md]: What happened to the periods at the end of sentences?

Comment (SmallJoker) [inline doc/client_lua_api.md]: This one is not a complete sentence as there's no personal pronoun. A period would be wrong here and in the most other descriptions. I guess just ignoring this is the easiest way.

**Comment by bigfoot547** [comment]
You are correct.

File: doc/client_lua_api.md (hunk 3 above)
Comment (bigfoot547) [inline doc/client_lua_api.md]: Yes, you are correct.

File: doc/client_lua_api.md [hunk 4]
```diff
@@ -621,6 +628,8 @@ Minetest namespace reference
   version entirely. To check for the presence of engine features, test
   whether the functions exported by the wanted features exist. For example:
   `if minetest.nodeupdate then ... end`.
+* `minetest.get_builtin_path()`
+    * Returns the path to builtin. Not guaranteed to return the absolute path.
```
Comment (ShadowNinja) [inline doc/client_lua_api.md]: This is really just an internal function and shouldn't be documented.

File: doc/lua_api.txt [hunk 5]
```diff
@@ -1931,7 +1931,10 @@ Helper functions
     * e.g. `string:split("a,b", ",") == {"a","b"}`
 * `string:trim()`
     * e.g. `string.trim("\n \t\tfoo bar\t ") == "foo bar"`
-* `minetest.pos_to_string({x=X,y=Y,z=Z}, decimal_places))`: returns `"(X,Y,Z)"`
+* `minetest.splittext(str, limit)`: returns a string
+    * Adds new lines to the string to keep it within the specified character limit
+    * limit: Maximal amount of characters in one line
```
Comment (ShadowNinja) [inline doc/lua_api.txt]: Isn't this just a hack that the main menu uses to work around line wrapping not being available?  I don't think it should really be exposed.
And if so, it should be `split_text`, or rather something more descriptive like `wrap_text`.

File: doc/lua_api.txt [hunk 6]
```diff
@@ -2027,6 +2030,8 @@ Helper functions
   version entirely. To check for the presence of engine features, test
   whether the functions exported by the wanted features exist. For example:
   `if minetest.nodeupdate then ... end`.
+* `minetest.get_builtin_path()`
+    * Returns the path to builtin. Not guaranteed to return the absolute path.
```
Comment (ShadowNinja) [inline doc/lua_api.txt]: Internal function.

File: doc/client_lua_api.md [hunk 7]
```diff
@@ -590,17 +590,24 @@ Helper functions
...
-* sep_is_pattern=false)`
+* `string.split(str, separator=",", include_empty=false, max_splits=-1, sep_is_pattern=false)`
     * If `max_splits` is negative, do not limit splits.
     * `sep_is_pattern` specifies if separator is a plain string or a pattern (regex).
     * e.g. `string:split("a,b", ",") == {"a","b"}`
 * `string:trim()`
     * e.g. `string.trim("\n \t\tfoo bar\t ") == "foo bar"`
+* `minetest.splittext(str, limit)`: returns a string
+    * Adds new lines to the string to keep it within the specified character limit
+    * limit: Maximal amount of characters in one line
```
Comment (ShadowNinja) [inline doc/client_lua_api.md]: See other comment.

File: doc/lua_api.txt (hunk 5 above)
Comment (SmallJoker) [inline doc/lua_api.txt]: I think the modders always will find an use for such a function. Added as `wrap_text`.

File: doc/client_lua_api.md (hunk 4 above)
Comment (red-001) [inline doc/client_lua_api.md]: Plus it will likely get removed for broken once sending mods from the server is added

**Comment by paramat** [comment]
Added +1 for ShadowNinja as his requests are attended to.

**Comment by paramat** [comment]
0120fe16a761f9e06c8c2877439db6a46d808143

**Pull request closed without merging**